### DISCUSSION TITLE
How to deal with column type needed situation?

### DISCUSSION THREAD

**Discussion by Ma233** [fastapi/sqlmodel#1517] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel, UniqueConstraint, select


class BookCollection(SQLModel, table=True):
    user_nid: int
    book_nid: int


UniqueConstraint(
    BookCollection.user_nid,  # Argument of type "int" cannot be assigned to parameter "columns" of type "str | Column[Any]" in function "__init__"
    BookCollection.book_nid,  # Argument of type "int" cannot be assigned to parameter "columns" of type "str | Column[Any]" in function "__init__"
    name="uidx_book_collection_user_nid_book_nid",
)

# Cannot access member "not_in" for type "int"
select(BookCollection).where(BookCollection.book_nid.not_in([1, 2, 3]))
```

### Description

Some APIs of sqlalchemy may still need a column type. Without that, a type checker will complain.
Currently, I'm using `type: ignore` to skip those.

### Operating System

Linux, macOS

### SQLModel Version

0.0.4

### Python Version

3.8.6

**Comment by ljluestc** [comment]
from sqlmodel import SQLModel, select, Column
from sqlalchemy import UniqueConstraint
from typing import List

class BookCollection(SQLModel, table=True):
    user_nid: int
    book_nid: int


# Define the table's UniqueConstraint using SQLAlchemy's Column objects
uidx_book_collection = UniqueConstraint(
    Column("user_nid", Integer),  # Specify the column type
    Column("book_nid", Integer),  # Specify the column type
    name="uidx_book_collection_user_nid_book_nid"
)

# Use a list to specify the column names for the "in_" condition
book_nids: List[int] = [1, 2, 3]
query = select(BookCollection).where(BookCollection.book_nid.in_(book_nids))

print(uidx_book_collection)
print(query)

**Comment by YuriiMotov** [comment, marked as answer]
In both cases you should use `col()` (`from sqlmodel import col`):

```py
from sqlmodel import (
    Field,
    SQLModel,
    UniqueConstraint,
    col,
    select,
)


class BookCollection(SQLModel, table=True):
    id: int = Field(primary_key=True)
    user_nid: int
    book_nid: int


UniqueConstraint(
    col(BookCollection.user_nid),
    col(BookCollection.book_nid),
    name="uidx_book_collection_user_nid_book_nid",
)

select(BookCollection).where(col(BookCollection.book_nid).not_in([1, 2, 3]))

```
